### DISCUSSION TITLE
15.0.0

### DISCUSSION THREAD

**Discussion by BasThomas** [Moya/Moya#2201] [Announcements]
# [15.0.0] - 2021-09-04

### Added
- Added `NetworkLoggerPlugin.default` and `NetworkLoggerPlugin.verbose` to conveniently access the default plugins. [#2095](https://github.com/Moya/Moya/pull/2095) by [@sunshinejr](https://github.com/sunshinejr).

### Changed
- **Breaking Change** Changed `Hashable` && `Equatable` implementation of `Endpoint` since it was returning false positives. [#2101](https://github.com/Moya/Moya/pull/2101) by [@sunshinejr](https://github.com/sunshinejr).
- **Breaking Change** `MultiPartFormData` is now `Hashable`. [#2101](https://github.com/Moya/Moya/pull/2101) by [@sunshinejr](https://github.com/sunshinejr).
- **Breaking Change** `AccessTokenPlugin` now uses `TargetType`, instead of `AuthorizationType`, in the closure to determine the token. Full `MultiTarget` integration added as well. [#2046](https://github.com/Moya/Moya/pull/2046) by [@Coder-ZJQ](https://github.com/Coder-ZJQ).
- `Target.sampleData` is now automatically implemented as `Data()` with default protocol extension. [#2015](https://github.com/Moya/Moya/pull/2015) by [jdisho](https://github.com/jdisho).
- **Breaking Change** Minimum version of `RxSwift` is now 6.0. [#2120](https://github.com/Moya/Moya/pull/2120) by [@peagasilva](https://github.com/peagasilva).
- Moya's Swift minimum version is now Swift 5.2. [#2120](https://github.com/Moya/Moya/pull/2120) by [@peagasilva](https://github.com/peagasilva).
- Moya now depends on the latest versions of RxSwift, ReactiveSwift & Alamofire. [#2197](https://github.com/Moya/Moya/pull/2197) by [@BasThomas](https://github.com/BasThomas).

### Fixed
- Fixed an issue where when using `trackInflights` option in certain circumstances would return a cached response for an endpoint that's not really the same. [#2101](https://github.com/Moya/Moya/pull/2101) by [@sunshinejr](https://github.com/sunshinejr).
- Fixed a crash where Combine Publisher would crash when using stubs.  [#2072](https://github.com/Moya/Moya/pull/2072) by [jshier](https://github.com/jshier).
- Fixed Unit Tests and CI. [#2187](https://github.com/Moya/Moya/pull/2187) by [OhKanghoon](https://github.com/OhKanghoon).
- Fixed a race condition that could prevent plugin's `willSend(_:target:)` from being fired. [#2192](https://github.com/Moya/Moya/pull/2192) by [anton-plebanovich](https://github.com/anton-plebanovich).

<hr /><em>This discussion was created from the release <a href='https://github.com/Moya/Moya/releases/tag/15.0.0'>15.0.0</a>.</em>

**Comment by hourksavet** [comment]
Hello, 
Maybe my wrong usage or your issue.
I am using **Maya 15.0.0** to request parameter by encoding `queryString`.

Here is my usage 
`return .requestParameters(parameters: ["topic": chatWith.rawValue, "type": "user", "client": ""], encoding: URLEncoding.queryString)`

`chatWith.rawValue = 2` but in request log show `chatWith.rawValue = 1`
If type case of `enum` is `String`, Its working fine.

Thank you!
